### DISCUSSION TITLE
Load all unloaded extension

### DISCUSSION THREAD

**Discussion by SichangHe** [neoclide/coc.nvim#4165] [Q&A] [answered]
Is there a way to _load all_ the _extensions_ that are _not loaded_ using a command or a function?
I use Dein to lazy load plugins based on filetypes, but when I open a file which should trigger a lazily loaded Coc extension plugin  by adding it to the runtimepath, Coc does not load that extension. For now, my workaround is to restart Coc client, which is inefficient. Is there a better way to do that?

**Comment by chemzqm** [comment]
What vim version?

**Reply by SichangHe** [reply]
```shell
❯ nvim --version
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@HMBRW-A-001-M1-004.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.7.2_1/share/nvim"

Run :checkhealth for more info
```

**Reply by chemzqm** [reply]
runtimepath change should be detected, the extension could be loaded but not activated.

**Reply by SichangHe** [reply]
Okay, so how do I activate them?

**Reply by SichangHe** [reply]
I can confirm that they would appear in `:CocList extensions`, but they are not activated.

**Reply by chemzqm** [reply]
Can't reproduce. What's your filetype and coc extension?

**Comment by SichangHe** [comment, marked as answer]
## Workaround
Currently, I use a "workspace"-like approach.
I load Coc extensions based on all the files in the current directory instead of only the file opened. Then, when the current directory changes I check and reload Coc if needed. So, Coc does not need to be reloaded when I open any other files in the current directory.

For an example, see [my lazy loaded nvim plugins config](https://github.com/SichangHe/.config/blob/main/nvim/dein_lazy.toml).
